### PR TITLE
feat(share-links): Add protobuf and encode/decode url data methods

### PR DESCRIPTION
**Pull request by borismelnik**
- Adds new module for `brotli` compression 
- Adds `encode/decode` methods for url data

Important changes:
- [x] Something worth noting for reviewers.

part of  #3622

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
Looks good. Maybe later when we have the full API we'd need a test that shows that the encode and decode work as expected

**Comment by borismelnik** [comment]
> Looks good. Maybe later when we have the full API we'd need a test that shows that the encode and decode work as expected

it will be in the next PR :)

**Review by MishkaRogachev — APPROVED** [review]

File: protocol/urls/urls.go [hunk 1]
```diff
@@ -293,3 +293,11 @@ func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
 		return previewData, fmt.Errorf("link %s isn't whitelisted. Hostname - %s", link, u.Hostname())
 	}
 }
+
+func EncodeDataURL(data []byte) (string, error) {
+	return encodeDataURL(data)
+}
+
+func DecodeDataURL(data string) ([]byte, error) {
```
Comment (ilmotta) [inline protocol/urls/urls.go:301]: Unit tests for the public functions `EncodeDataURL` and `DecodeDataURL` seem appropriate.

File: protocol/urls/urls.go [hunk 2]
```diff
@@ -293,3 +293,11 @@ func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
 		return previewData, fmt.Errorf("link %s isn't whitelisted. Hostname - %s", link, u.Hostname())
 	}
 }
+
```
Comment (ilmotta) [inline protocol/urls/urls.go:296]: Worth mentioning that I created a different package `linkpreview` in `protocol/` for dealing specifically with link previews and  the idea was to delete the `urls` package entirely since it doesn't fulfill clients' needs anymore.

I think the PR can be merged using the `urls` package, but I think we should refactor soon(ish) to use the `Unfurler` interface defined in `protocol/linkpreview` so that we don't grow the package that was supposed to be deleted (when the new link preview implementation is more complete featurewise).

**OR forget what I just said above**, and we can use/think of a completely different package and interface to deal with "domain" data (community, channel, user) because these can be seen as significantly different from unfurleable data from websites. So maybe using the package `urls` is okay. Food for thought :)

File: protocol/protobuf/url_data.proto [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ // Community, Channel, or User
+ bytes content = 1;
+}
+
+// Field on CommunityDescription, CommunityChat and ContactCodeAdvertisement
+message URLParams {
+ string encoded_url_data = 1;
+ // Signature of encoded URL data
+ string encoded_signature = 2;
+}
```
Comment (ilmotta) [inline protocol/protobuf/url_data.proto:39]: It's generally accepted that we should end files with *proper* lines. If a reference helps, according to the POSIX standard, proper lines should have a terminating newline character. If you allow me, I recommend configuring your text editor of choice to do this for you. All of them have this feature.

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

File: protocol/urls/urls.go (hunk 1 above)
Comment (Samyoul) [reply protocol/urls/urls.go:301]: Not wanting to dogpile you @borismelnik, I noticed your comment https://github.com/status-im/status-go/pull/3650#issuecomment-1599276139 , but we should never introduce code into this repo without proving that it does what it is expected to do.

File: protocol/protobuf/url_data.proto [hunk 4]
```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
```
Comment (Samyoul) [inline protocol/protobuf/url_data.proto:1]: I notice that these protobufs are introduced but are not used anywhere in this PR. Am I correct in thinking that they perhaps belong in another PR?

Comment (borismelnik) [reply protocol/protobuf/url_data.proto:1]: @Samyoul Yes, this PR is a general PR for next 2 PRs

File: protocol/urls/urls.go (hunk 1 above)
Comment (borismelnik) [reply protocol/urls/urls.go:301]: okay! I will add the tests in this PR

Comment (Samyoul) [reply protocol/urls/urls.go:301]: Thank you very much, I really appreciate that 🙏

**Comment by borismelnik** [comment]
in fact of https://github.com/status-im/status-go/pull/3665
